src/sqlite3gen.cpp

```diff
@@ -23,77 +23,145 @@
   },
-  { "innerclass",
-    "CREATE TABLE IF NOT EXISTS innerclass (\n"
+  { "contains",
+    "CREATE TABLE IF NOT EXISTS contains (\n"
+      "\t-- inner/outer relations (file, namespace, dir, class, group, page)\n"
       "\trowid        INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,\n"
-      "\trefid        INTEGER NOT NULL,\n"
-      "\tprot         INTEGER NOT NULL,\n"
-      "\tname         TEXT NOT NULL\n"
-      ");"
+      "\tinner_rowid  INTEGER NOT NULL REFERENCES compounddef,\n"
+      "\touter_rowid  INTEGER NOT NULL REFERENCES compounddef\n"
+    ");"
   },
-  { "files",
-    "CREATE TABLE IF NOT EXISTS files (\n"
-      "\t-- Names of source files and includes.\n"
-      "\tname         TEXT PRIMARY KEY NOT NULL\n"
-      ");"
+  /* TODO: Path can also share rowids with refid/compounddef/def. (It could
+   *       even collapse into that table...)
+   *
+   * I took a first swing at this by changing insertPath() to:
+   * - accept a FileDef
+   * - make its own call to insertRefid
+   * - return a refid struct.
+   *
+   * I rolled this back when I had trouble getting a FileDef for all types
+   * (PageDef in particular).
+   *
+   * Note: all colums referencing path would need an update.
+   */
+  { "path",
+    "CREATE TABLE IF NOT EXISTS path (\n"
+      "\t-- Paths of source files and includes.\n"
+      "\trowid        INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,\n"
+      "\ttype         INTEGER NOT NULL, -- 1:file 2:dir\n"
+      "\tlocal        INTEGER NOT NULL,\n"
+      "\tfound        INTEGER NOT NULL,\n"
+      "\tname         TEXT NOT NULL\n"
+    ");"
   },
-  { "refids",
-    "CREATE TABLE IF NOT EXISTS refids (\n"
-      "\trefid        TEXT PRIMARY KEY NOT NULL\n"
-      ");"
+  { "refid",
+    "CREATE TABLE IF NOT EXISTS refid (\n"
+      "\t-- Distinct refid for all documented entities.\n"
+      "\trowid        INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,\n"
+      "\trefid        TEXT NOT NULL UNIQUE\n"
+    ");"
   },
   { "xrefs",
     "CREATE TABLE IF NOT EXISTS xrefs (\n"
-      "\t-- Cross reference relation.\n"
+      "\t-- Cross-reference relation\n"
+      "\t-- (combines xml <referencedby> and <references> nodes).\n"
       "\trowid        INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,\n"
-      "\trefid_src    INTEGER NOT NULL, -- referrer id.\n"
-      "\trefid_dst    INTEGER NOT NULL, -- referee id.\n"
-      "\tid_file      INTEGER NOT NULL, -- file where the reference is happening.\n"
-      "\tline         INTEGER NOT NULL, -- line where the reference is happening.\n"
-      "\tcolumn       INTEGER NOT NULL  -- column where the reference is happening.\n"
-      ");\n"
-    "CREATE UNIQUE INDEX idx_xrefs ON xrefs\n"
-      "\t(refid_src, refid_dst, id_file, line, column);"
+      "\tsrc_rowid    INTEGER NOT NULL REFERENCES refid, -- referrer id.\n"
+      "\tdst_rowid    INTEGER NOT NULL REFERENCES refid, -- referee id.\n"
+      "\tcontext      TEXT NOT NULL, -- inline, argument, initializer\n"
```

Comment:
Local neologism. Attempting to validate the sqlite3 output against the XML output revealed that there were 3 different contexts that resulted in the documentation of an xref (inline, argument, and initializer).